### PR TITLE
Greeting Message

### PR DESCRIPTION
**Pull request by ShahierNashaat**
### Pull request for review create rails-react app using Ruby on Rails, React, and Redux 

- Create a new Rails app called 'hello-rails-react'.
 - Initialize your project with Git.
 - Set up your project with `webpacker` the `react-rails` gems as describe in the previous lesson's tutorial.
 - Make sure that your project has a Postgres database set up. 
 - Create a table for storing your messages and create 5 different greetings. 
 - Create an API endpoint that selects a random greeting from your table (you will need a controller with an action and Rails route).
 - Create a static view that will be the root of your app.

---------

- Create the `App` component with react-router and render it in your static view.
- Create the `Greeting` component that will display a greeting. Set it up as a route in your App component.
- Create a store, an action and a reducer that will connect to you API endpont to get the random greeting.
- Display the random greeting in your Greeting component.

### CODE REVIEW

File: app/javascript/components/Greeting.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { fetchmessage } from '../redux/message/api';
```
Comment (ahmetbozaci) [inline app/javascript/components/Greeting.js]: - [ ] You should update the package.json file. Many dependencies are missing. That is why ı am getting a few modules not found an error. One example is you are using react-redux here but it is not in the package.json file. Please check other dependencies.

- [ ] `fetchmessage` Path is wrong that's why it is giving an error. Please fix it.
image

File: app/javascript/components/Greeting.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+const Greeting = () => {
+  const message = useSelector((state) => state.messageReducer.message);
+  // console.log('message => ', message);
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(fetchmessage());
+  }, []);
```
Comment (ahmetbozaci) [inline app/javascript/components/Greeting.js:10]: - [ ] After fixing the path in the Greeting file ı am getting this error. Please fix it.

image

File: config/routes.rb [hunk 3]
```diff
@@ -1,6 +1,9 @@
 Rails.application.routes.draw do
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 
-  # Defines the root path route ("/")
-  # root "articles#index"
+  namespace :api, defaults: { format: 'json' } do
+    get '/messages', to: 'messages#index'
+  end
+
+  root 'static#index'
```
Comment (ahmetbozaci) [inline config/routes.rb]: - [ ] Something is wrong in your route or `fetchmessage` function in `redux/reducers/messages` file. Please fix it.

image

File: config/database.yml [hunk 4]
```diff
@@ -17,6 +17,10 @@
 default: &default
   adapter: postgresql
   encoding: unicode
+  username: postgres
+  password: 1234
+  host: localhost
+  port: 5432
```
Comment (ahmetbozaci) [inline config/database.yml:23]: ### OPTIONAL
- [ ] It is always good practice not to expose your database login information. Consider using the env file for all sensitive information about your application.

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# STATUS: Changes Required  ♻️

### Hello @ShahierNashaat  ✋🏼 
#### Your project is great. You are almost done. However, there are some issues you should fix. Please fix them then your project will be approved.


### ♻️ Request changes ♻️

- [ ] _Check the comments under the review._

___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___
- You have one commit for everything. For best practices, you should commit often. [Git Commit Best Practices](https://www.perforce.com/blog/vcs/git-best-practices-git-commit)

almost

___

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_


_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mentioning me as [ahmetbozaci](https://github.com/ahmetbozaci), I will receive a notification and will respond._
___
[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Review by dasileker — APPROVED** [review]
### **Hi @ShahierNashaat,** 

### **Congratulations! 🎉**

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 : 
 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**